watertap/unit_models/tests/test_electroNP_ZO.py

```diff
@@ -0,0 +1,263 @@
+#################################################################################
+# WaterTAP Copyright (c) 2020-2023, The Regents of the University of California,
+# through Lawrence Berkeley National Laboratory, Oak Ridge National Laboratory,
+# National Renewable Energy Laboratory, and National Energy Technology
+# Laboratory (subject to receipt of any required approvals from the U.S. Dept.
+# of Energy). All rights reserved.
+#
+# Please see the files COPYRIGHT.md and LICENSE.md for full copyright and license
+# information, respectively. These files are also available online at the URL
+# "https://github.com/watertap-org/watertap/"
+#################################################################################
+
+import pytest
+from pyomo.environ import (
+    ConcreteModel,
+    assert_optimal_termination,
+    value,
+    Var,
+    log10,
+    units,
+)
+from pyomo.network import Port
+from idaes.core import (
+    FlowsheetBlock,
+    MaterialBalanceType,
+    EnergyBalanceType,
+    MomentumBalanceType,
+)
+from watertap.unit_models.electroNP_ZO import ElectroNPZO
+from watertap.property_models.activated_sludge.modified_asm2d_properties import (
+    ModifiedASM2dParameterBlock,
+)
+from watertap.property_models.activated_sludge.modified_asm2d_reactions import (
+    ModifiedASM2dReactionParameterBlock,
+    ModifiedASM2dReactionBlock,
+)
+from idaes.core.solvers import get_solver
+from idaes.core.util.model_statistics import (
+    degrees_of_freedom,
+    number_variables,
+    number_total_constraints,
+    number_unused_variables,
+)
+from idaes.core.util.testing import initialization_tester
+from idaes.core.util.scaling import (
+    calculate_scaling_factors,
+    unscaled_variables_generator,
+    unscaled_constraints_generator,
+    badly_scaled_var_generator,
+)
+from pyomo.util.check_units import assert_units_consistent
+from idaes.core import UnitModelCostingBlock
+from watertap.costing import WaterTAPCosting
+
+# -----------------------------------------------------------------------------
+# Get default solver for testing
+solver = get_solver()
+
+
+class TestElectroNP:
+    @pytest.fixture(scope="class")
+    def ElectroNP_frame(self):
+        m = ConcreteModel()
+        m.fs = FlowsheetBlock(dynamic=False)
+
+        m.fs.properties = ModifiedASM2dParameterBlock(
```

Comment:
Hmmm.... I just thought about this, and there is a chance, due to how we've compartmentalized the property and reaction models, that some expressions/additional variables/additional constraints implemented in the reaction block _could_ be desired in the unit model (but not sure yet since I didn't think about it deeply). Maybe this isn't an issue yet, and my concern is probably more relevant to our ADM1 formulation rather than ASM2D, but let's keep this in mind since we expect to make additions to the modified ASM2d property/reaction package.

Reply:
Yeah, this is a good point, some additional variables can be added to the unit model instead of properties model. Just for modified ASM2D package, we might need to keep the additonal_solute config for now.